pytorch_transformers/tokenization_utils.py

```diff
@@ -520,12 +520,29 @@ def _convert_token_to_id(self, token):
         raise NotImplementedError
 
 
-    def encode(self, text):
+    def encode(self, *text, cls_token_at_end=False, double_sep_token=False, no_sep_cls_tokens=False):
         """ Converts a string in a sequence of ids (integer), using the tokenizer and vocabulary.
         
         Same doing ``self.convert_tokens_to_ids(self.tokenize(text))``.
         """
-        return self.convert_tokens_to_ids(self.tokenize(text))
+
+        if len(text) == 1:
+            return self.convert_tokens_to_ids(self.tokenize(text[0]), no_sep_cls_tokens=no_sep_cls_tokens)
+
+        if len(text) > 2:
+            logger.warning("Tokenization currently only supports sentence pairs. Ignoring every string following the "
+                           "initial two.")
+
+        first_sentence_tokens = [self._convert_token_to_id(token) for token in self.tokenize(text[0])]
+        second_sentence_tokens = [self._convert_token_to_id(token) for token in self.tokenize(text[1])]
+        sep = [self._convert_token_to_id(self.sep_token)]
+        cls = [self._convert_token_to_id(self.cls_token)]
+        n_sep_token = 2 if double_sep_token else 1
+
+        tokens = first_sentence_tokens + sep * n_sep_token + second_sentence_tokens + sep
+        tokens = (tokens + cls) if cls_token_at_end else (cls + tokens)
+
+        return tokens
```

Comment:
@thomwolf Would like to know what you think about this. It doesn't break the current implementation. I have not yet added tests for sentence-pairs but will do soon.

Reply:
I like the idea of counting the number of positional arguments but I'm not a fan of all the keywords arguments.

We define these options today but new models will likely require different options and we'll keep making this more and more complex. Plus, the end-user doesn't really care how the special tokens are precisely setup. The user either (i) wants to have the right special tokens for the right model without having to know the inside or (ii) wants to add the special token him-self, in which case we should just encode the input and let the user add the special tokens him/her-self.

What about having a single boolean argument `add_special_tokens` that will choose between these two behaviors and have a workflow like this:
- we encode (without special tokens) all the texts,
- if `add_special_tokens==True` we call a method implemented for each model `add_special_tokens_single_sentence` or `add_special_tokens_sentences_pair` depending on the number of texts inputs.

We can define default behavior for the `add_special_tokens_single_sentence` and `add_special_tokens_sentences_pair` method at the base tokenizer level that will basically do nothing and then do all the specific things like `double_sep_token` in the model-secific tokenizers.

Reply:
Good points; this looks to be more modular than what I came up with, as well as more future-proof.